modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,68 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  it('TINY-8641: Dispatch change and set editor dirty if the current content and the undoManager history are incoherent', () => {
+    const editor = hook.editor();
+
+    let changeEventCounter = 0;
+    let currentChangeEvent: { level: string; lastLevel: string };
+
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const assertChangeEvent = (event: { level: string; lastLevel: string }) => {
+      assert.equal(event.level, initialContent, 'Level has not the expected content');
+      assert.equal(event.lastLevel, manualModifiedLevel, 'Last level has not the expected content');
+    };
+
+    editor.resetContent(initialContent);
+    const onChange = (e: EditorEvent<{
+      level: UndoLevel;
+      lastLevel: UndoLevel;
+    }>) => {
+      changeEventCounter++;
+      currentChangeEvent = { level: e.level.content, lastLevel: e.lastLevel.content };
+    };
+
+    editor.on('change', onChange);
+
+    assert.equal(changeEventCounter, 0, '0 event should be detected at start');
+    editor.undoManager.dispatchIfChanged();
+    assert.equal(changeEventCounter, 0, '0 event should be detected if content is coherent with the undoManager history');
+
+    Arr.last(editor.undoManager.data).each((lastLevel) => {
+      lastLevel.content = manualModifiedLevel;
+    });
+
+    assert.equal(editor.isDirty(), false, 'Editor should not be dirty before dispatchIfChanged');
```

Comment:
Please use the `assert.isFalse` function instead for these type of assertions. There's also `assert.isTrue` for the cases below where you do a truthy check instead.

- [ ] Use the `isTrue` or `isFalse` assertions instead of manually doing it via `equal`

Reply:
done